include/boost/geometry/formulas/gnomonic_intersection.hpp

```diff
@@ -0,0 +1,147 @@
+// Boost.Geometry
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Adam Wulkiewicz, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_GNOMONIC_INTERSECTION_HPP
+#define BOOST_GEOMETRY_FORMULAS_GNOMONIC_INTERSECTION_HPP
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/cs.hpp>
+
+#include <boost/geometry/arithmetic/cross_product.hpp>
+#include <boost/geometry/formulas/gnomonic_spheroid.hpp>
+#include <boost/geometry/geometries/point.hpp>
+#include <boost/geometry/util/math.hpp>
+
+
+namespace boost { namespace geometry { namespace formula
+{
+
+/*!
+\brief The intersection of two geodesics using spheroidal gnomonic projection
+       as proposed by Karney.
+\author See
+    - Charles F.F Karney, Algorithms for geodesics, 2011
+      https://arxiv.org/pdf/1109.4448.pdf
+    - GeographicLib forum thread: Intersection between two geodesic lines
+      https://sourceforge.net/p/geographiclib/discussion/1026621/thread/21aaff9f/
```

Comment:
The `\author` is the author of the papers/formulas, not author of the code. I've taken this description template from Andoyer and Vincenty which was first formulas implemented in BG. And there the `\author` mentions papers containing the original formulas.

I haven't used the code of GeographicLib directly which indeed is MIT licensed. However indeed I've seen the code and used it as a complementary source of formulas to the paper because some things were not obvious in the paper (in particular in case of differential quantities and gnomonic projection).

This particular formula (intersection using gnomonic projection) is not implemented in GeographicLib. A simple example was attached by Karney in the forum thread but I haven't used it directly either (plus there was no license there).